src/modules/inventing/QuickInvent.svelte

```diff
@@ -0,0 +1,61 @@
+<script>
+  import daDoInvent from '../_dataAccess/daDoInvent';
+  import { sendEvent } from '../support/fshGa';
+
+  export let max;
+  export let recipeID;
+  let amountToInvent = 1;
+  let invResults = '';
+  let results = [];
+
+  function maxInvent() {
+    amountToInvent = max;
+    sendEvent('inventing', 'maxInventButton');
+  }
+
+  async function quickInvent() {
+    sendEvent('inventing', 'quickInvent');
+    results = [];
+    invResults = '';
+    if (!amountToInvent) { return; }
+    invResults = `Inventing ${amountToInvent} Items`;
+    const requests = Array(amountToInvent).fill(recipeID);
+    requests.reduce(async (prev, recipe) => {
+      await prev;
+      return daDoInvent(recipe).then((json) => {
```

Comment:
then -> await

Reply:
Changing the `await` to a `then` like you did in `trade/oneByOne.js` doesn't seem to work here. Perhaps it's something about svelte I haven't figured out yet, but if I don't await the previous promise, I can't modify the results array.

Reply:
my comment was unclear.
i should have said ```daDoInvent(recipe).then(``` should be an await i think.
have a look at ```trade/oneByOne.js``` in master branch.
```suggestion
      const data = await prev;
      if (!data || data.s) {
        const json = await daDoInvent(recipe);
        results = [...results, json];
        return json;
      }
      return data;
```